### PR TITLE
Build less of libs for NativeAOT testing

### PR DESCRIPTION
**Pull request by MichalStrehovsky**
We don't care about any of the System.Speech, Microsoft.Extensions.*,... stuff.

Made possible with #89005.

Cc @dotnet/ilc-contrib

### CODE REVIEW

File: eng/pipelines/runtime.yml [hunk 1]
```diff
@@ -208,7 +208,7 @@ extends:
           jobParameters:
             timeoutInMinutes: 120
             nameSuffix: NativeAOT
-            buildArgs: -s clr.aot+host.native+libs -rc $(_BuildConfig) -lc Release -hc Release
+            buildArgs: -s clr.aot+host.native+libs.sfx+libs.native -rc $(_BuildConfig) -lc Release -hc Release
```
Comment (ViktorHofer) [inline eng/pipelines/runtime.yml:211]: nit: build libs.native before libs.sfx. That's the expected order...

**Comment by MichalStrehovsky** [comment]
Looking at the test failures, looks like EventPipe testing relies on some Microsoft.Extensions crap and since we test EventPipe as part of smoke tests now, we need the whole product. This would have been enough 6 months ago when we didn't have EventPipe, but not anymore. Oh well :'(.

**Comment by MichalStrehovsky** [comment]
I'm still going to use this in my personal workflows though because I don't run event pipe testing, so @ViktorHofer your https://github.com/dotnet/runtime/pull/89005 is still super useful.

**Comment by ViktorHofer** [comment]
As discussed offline with Michal, this would work just fine if the runtime tests in question could use ProjectReferences and bring in the required out-of-band libraries that are needed by the specific tests. Given that the current infra around runtime tests is fragile around PackageReferences/ProjectReferences we will not pursue this further at this point. cc @jkoritzinsky

**Comment by jkoritzinsky** [comment]
Once we do part two of test consolidation, we can revisit this.

**Pull request closed without merging**